metapool/sample_sheet.py

```diff
@@ -1753,7 +1766,7 @@ def _get_contains_replicates_value(sheet):
     # to be True or False. If some projects are True while others are
```

Comment:
My approach in this refactor was to NOT remove **any** existing functions. Thus, every existing function call, no matter where it may be buried in the various test or repos (which I am definitely not familiar with all of!), should continue to work, even if the work is now just being forwarded to a new function.  Basically, I'm trying not to rock the boat *too* much.  Given this, I didn't remove or rename `sheet_needs_demuxing`.  

In the future, we might change this.  We would need to take into account that `sheet_needs_demuxing` is tolerant of the lack of CONTAINS_REPLICATES_KEY, whereas `_get_contains_replicates_value` (and the other place outside of `sheet_needs_demuxing` where `_get_contains_replicates_value` is used) are NOT tolerant of that and need to throw errors.  Still, if we definitely want to combine the two, I'm sure we could find a way to switch on this.

I used the `_` to indicate the new function I added is supposed to be private to this module.  I did this because right now it *isn't* being used anywhere outside this module, and I think it should stay that way until we decide if we want ;)

Reply:
Not removing existing functions in this PR is reasonable. Yeah I think we should combine the two at some point.